File: src/main/java/cz/vhromada/catalog/web/mapper/GameMapper.java
package cz.vhromada.catalog.web.mapper;

import cz.vhromada.catalog.entity.Game;
import cz.vhromada.catalog.web.fo.GameFO;

import org.mapstruct.Mapper;

/**
 * An interface represents mapper for games.
 *
 * @author Vladimir Hromada
 */
@Mapper
public interface GameMapper {

    /**
     * Returns FO for game.
     *
     * @param source game
     * @return FO for game
     */
    GameFO map(Game source);

    /**
     * Returns game.
     *
     * @param source FO for game
     * @return game
     */
    Game mapBack(GameFO source);

}

File: src/main/kotlin/cz/vhromada/catalog/web/mapper/GameMapper.kt
package cz.vhromada.catalog.web.mapper

import cz.vhromada.catalog.entity.Game
import cz.vhromada.catalog.web.fo.GameFO

/**
 * An interface represents mapper for games.
 *
 * @author Vladimir Hromada
 */
interface GameMapper {

    /**
     * Returns FO for game.
     *
     * @param source game
     * @return FO for game
     */
    fun map(source: Game): GameFO

    /**
     * Returns game.
     *
     * @param source FO for game
     * @return game
     */
    fun mapBack(source: GameFO): Game

}
